public/data/cpp.json

```diff
@@ -16,5 +16,90 @@
         "author": "Vaibhav-kesarwani"
       }
     ]
+  },
+  {
+    "categoryName": "Array Manipulation",
+    "snippets": [
+      {
+        "title": "Find Minimum and Maximum Values in an Array",
+        "description": "Finds the minimum and maximum values in an array using only a single loop.",
+        "code": [
+          "void findMinMax(const int arr[], int size, int& min, int& max) {",
+          "    min = arr[0];",
+          "    max = arr[0];",
+          "    for (int i = 1; i < size; i++) {",
+          "        if (arr[i] < min) {",
+          "            min = arr[i];",
+          "        } else if (arr[i] > max) {",
+          "            max = arr[i];",
+          "        }",
+          "    }",
+          "}"
+        ],
+        "tags": ["cpp", "array", "loop", "min-max", "min", "max"],
+        "author": "shovan04"
+      },
+      {
+        "title": "Find Second Largest Element in an Array",
+        "description": "Finds the second largest element in an array using only a single loop.",
+        "code": [
+          "int findSecondLargest(const int arr[], int size) {",
+          "    int largest = INT_MIN;",
+          "    int secondLargest = INT_MIN;",
+          "    for (int i = 0; i < size; i++) {",
+          "        if (arr[i] > largest) {",
+          "            secondLargest = largest;",
+          "            largest = arr[i];",
+          "        } else if (arr[i] < largest && arr[i] > secondLargest) {",
+          "            secondLargest = arr[i];",
+          "        }",
+          "    }",
+          "    return secondLargest;",
+          "}"
+        ],
+        "tags": ["cpp", "array", "loop", "Largest", "Second Largest"],
+        "author": "shovan04"
+      },
+      {
+        "title": "Sort Array in Descending Order",
+        "description": "Sorts an array in descending order using the Bubble Sort algorithm.",
+        "code": [
+          "void bubbleSortDescending(int arr[], int size) {",
+          "    for (int i = 0; i < size - 1; i++) {",
+          "        for (int j = 0; j < size - i - 1; j++) {",
+          "            if (arr[j] < arr[j + 1]) {",
+          "                std::swap(arr[j], arr[j + 1]);",
+          "            }",
+          "        }",
+          "    }",
+          "}"
+        ],
+        "tags": ["cpp", "array", "bubble-sort", "descending", "swap"],
+        "author": "shovan04"
+      },
+      {
+        "title": "Find Common Elements in Two Arrays",
+        "description": "Finds the common elements in two arrays using a single loop and without using any additional data structures.",
+        "code": [
+          "#include <climits>",
+          "",
+          "",
+          "void findCommonElements(const int arr1[], int size1, int arr2[], int size2, int& commonCount) {",
```

Comment:
Not a fan of this function too,
It doesn't return the common elements but the number of common elements.
It also modifies one the array which can be non desired behavior...